File: experiments-local/experiment-17/runs/agent=omp_language=csharp_model=openrouter/anthropic/claude-opus-4.8_tooling=none/rep1/src/BrazilianSoccer.Core/Competitions.cs
// -----------------------------------------------------------------------------
// File: Competitions.cs
// Project: BrazilianSoccer.Core
//
// Context:
//   Display names and free-text parsing for the Competition enum. Tool callers
//   (and ultimately an LLM) pass competition names as loose strings like
//   "Brasileirao", "serie a", "Copa do Brasil", "libertadores". This central
//   parser keeps that fuzzy mapping in one place instead of scattering string
//   comparisons through the query services.
// -----------------------------------------------------------------------------

using BrazilianSoccer.Core.Models;

namespace BrazilianSoccer.Core;

/// <summary>Display names and fuzzy parsing for <see cref="Competition"/>.</summary>
public static class Competitions
{
    /// <summary>Human-readable name for a competition.</summary>
    public static string DisplayName(Competition c) => c switch
    {
        Competition.BrasileiraoSerieA => "Brasileirão Série A",
        Competition.BrasileiraoSerieB => "Brasileirão Série B",
        Competition.BrasileiraoSerieC => "Brasileirão Série C",
        Competition.CopaDoBrasil => "Copa do Brasil",
        Competition.Libertadores => "Copa Libertadores",
        _ => "Unknown competition",
    };

    /// <summary>
    /// Parses a loose competition string. Returns null when the text is empty
    /// (meaning "all competitions") and <see cref="Competition.Unknown"/> when the
    /// text is non-empty but unrecognised (so callers can report a bad filter).
    /// </summary>
    public static Competition? Parse(string? text)
    {
        if (string.IsNullOrWhiteSpace(text))
            return null;

        var t = TeamName.MatchKey(text); // reuse accent/case folding

        if (t.Contains("libertadores"))
            return Competition.Libertadores;
        if (t.Contains("copa do brasil") || t == "cup" || t.Contains("copa brasil"))
            return Competition.CopaDoBrasil;
        if (t.Contains("serie b") || t.Contains("serie 2") || t.Contains("segunda"))
            return Competition.BrasileiraoSerieB;
        if (t.Contains("serie c") || t.Contains("serie 3") || t.Contains("terceira"))
            return Competition.BrasileiraoSerieC;
        if (t.Contains("serie a") || t.Contains("brasileirao") || t.Contains("brasileiro")
            || t.Contains("campeonato brasileiro"))
            return Competition.BrasileiraoSerieA;

        return Competition.Unknown;
    }
}

File: experiments-local/experiment-mu-glm53-provider/runs/agent=oc-fireworks_language=python_model=openrouter/z-ai/glm-5.3_tooling=none/rep2/brazilian_soccer/competitions.py
"""Canonical competition ids, display names and query aliases."""

from __future__ import annotations

from .normalize import fold

COMPETITIONS: dict[str, str] = {
    "brasileirao": "Brasileirão Serie A",
    "serie_b": "Brasileirão Serie B",
    "serie_c": "Brasileirão Serie C",
    "copa_do_brasil": "Copa do Brasil",
    "libertadores": "Copa Libertadores",
}

_LEAGUES = {"brasileirao", "serie_b", "serie_c"}

COMP_ALIASES: dict[str, str] = {}
for _key in COMPETITIONS:
    _display = COMPETITIONS[_key]
    COMP_ALIASES[fold(_key)] = _key
    COMP_ALIASES[fold(_display)] = _key
COMP_ALIASES.update(
    {
        fold("serie a"): "brasileirao",
        fold("seriea"): "brasileirao",
        fold("série a"): "brasileirao",
        fold("brasileirao serie a"): "brasileirao",
        fold("campeonato brasileiro"): "brasileirao",
        fold("serieb"): "serie_b",
        fold("serie b"): "serie_b",
        fold("seriec"): "serie_c",
        fold("serie c"): "serie_c",
        fold("copa"): "copa_do_brasil",
        fold("cup"): "copa_do_brasil",
        fold("brazilian cup"): "copa_do_brasil",
        fold("copa brasil"): "copa_do_brasil",
        fold("libertadores"): "libertadores",
        fold("conmebol libertadores"): "libertadores",
        fold("copa libertadores"): "libertadores",
    }
)


def resolve_competition(query: str | None) -> str | None:
    """Resolve a user-supplied competition name to a canonical id."""
    if not query:
        return None
    key = fold(query)
    if key in COMPETITIONS:
        return key
    return COMP_ALIASES.get(key)


def is_league(competition: str) -> bool:
    return competition in _LEAGUES
